src/Translatable/Translatable.php

```diff
@@ -458,4 +459,35 @@ public function __isset($key)
     {
         return $this->isTranslationAttribute($key) || parent::__isset($key);
     }
+
+    public function getTranslationChanges(): array
+    {
+        switch (config('translatable.rule_factory.format')) {
+            case RuleFactory::FORMAT_KEY:
+                $replacementFunc = function (string $key, string $locale): string {
+                    return $key . ':' . $locale;
+                };
+                break;
+            default:
+                $replacementFunc = function (string $key, string $locale): string {
+                    return $locale . '.' . $key;
+                };
+        }
+        $translationChanges = array();
+        foreach ($this->translations as $translation) {
+            $locale = $translation->locale;
+            foreach ($translation->getChanges() as $attribute => $value) {
+                $translationChanges[$replacementFunc($attribute, $locale)] = $value;
```

Comment:
Sorry for the confusion. Now I get your point.
I assumed that both formats would return a two-dimensional array.

My experience made me learn Postel's Law the hard way, so it feels kind of error prone to me to **return** different **types** depending on a config setting.
Correct me if I'm wrong but until now the format config only determined how **input** was interpreted (form field names in request). My initial solution had the same smell, at least the types were identical (only property/field names were composed depending on config) and I chose it for the simplicity it provided for the `hasTranslationChanged()` method.

I agree with you that returning a two-dimensional array would open further opportunities.
It would also be closer to the model's internal state (for both format options).

So I suggest to return a two-dimensional array here for both formats and maybe respect the format config in method `hasTranslationChanged()` where the user might **input** formatted(?) attribute names.
This would even further simplify the methods code:

```suggestion
            $changes = $translation->getChanges();
            if (!empty($changes)) {
                $translationChanges[$translation->locale] = $changes;
            }
```

Reply:
Sounds good. ✅

Less code to maintain and I never liked the `name:de` format. 😅

For the has changed method I would still prefer the two arguments so attribute(s) and locale separated.
If you want to allow the formatted strings in the attributes and parse them corresponding is up to you. The methods should be there (same as used in `fill()`.